### PR TITLE
Made server able to receive multiple clients. Added exit command for server and client.

### PR DESCRIPTION
**Pull request by hvostov**
With this adding server accepts multiple clients using select function. It works in infinity loop until receives a "exit" command from std::cin. Also added one thread for receiving this "exit" command. When Server accepts a client, we create an unordered map , that matches socket with username. We printing in command line, when user "username" connected or disconnected. 
Added same "exit" command for Client.

### CODE REVIEW

File: Server/src/server.cpp [hunk 1]
```diff
@@ -455,29 +449,25 @@ int Server::handleLsCommand(char *command) {
         std::string directoryContents = "Directory listing for " + cwd + "\n";
 
         // directory_iterator is used to traverse all the entries of directory
-        for (const auto & entry : std::filesystem::directory_iterator(cwd)) {
+        for (const auto& entry : std::filesystem::directory_iterator(cwd)) {
             directoryContents += entry.path().filename().string() + "\n";
         }
 
-        directoryContents += '\f';
```
Comment (Shu-AFK) [inline Server/src/server.cpp:462]: Why did you remove this?

File: Server/src/server.cpp [hunk 2]
```diff
@@ -768,49 +765,114 @@ int Server::shiftStrLeft(char *str, int num) {
...
 }
 
 /**
- * @brief Runs the server and continuously receives and handles commands from the client
+ * @brief Receives the command "exit" from std::cin to stop server
  *
  * @details
- * The run() function is responsible for running the server and continuously receiving and handling commands from the client.
- * It uses a do-while loop to keep receiving commands until it receives a return value of 1 or 2 from the handleCommand() function.
+ * Runs in its own thread and closes all connected sockets
```
Comment (Shu-AFK) [inline Server/src/server.cpp:782]: could you add an @parameter to the comment?

File: Server/src/server.cpp [hunk 3]
```diff
@@ -1660,30 +1729,33 @@ int Server::addStartup() {
...
-    if(space_counter != 1) return -1;
+    if (space_counter != 1) return -1;
 
     int res = auth(username, password);
-    if(res == -1) return -1;
+    if (res == -1) return -1;
 
-    if(handleSend("valid") != 0) return -1;
+    userMap[LastSock] = username;
+    std::cout << "Accepted new client. Username: " << username << std::endl;
```
Comment (Shu-AFK) [inline Server/src/server.cpp:1756]: Could you log this as well? Makes it easier to see where something gooes wrong if it does

File: Server/src/server.cpp (hunk 1 above)
Comment (Shu-AFK) [reply Server/src/server.cpp:462]: Also the lines below that line

Comment (hvostov) [reply Server/src/server.cpp:462]: this is the point of double \f.
below, just because everywhere else it was used handleSend command, but in this place - winsock send.

File: Server/src/server.cpp (hunk 2 above)
Comment (Shu-AFK) [reply Server/src/server.cpp:782]: Just commit to your branch and it gets added to this pr

File: Server/src/server.cpp (hunk 1 above)
Comment (Shu-AFK) [reply Server/src/server.cpp:462]: Alright, I was probably too tired yesterday to actually look at the code lmao
